### ISSUE TITLE
Can not handle `(group).[foo].md`  correctly

### ISSUE THREAD

**Issue by azu** [secretlint/secretlint#1057]
It is a bug:

```
$ secretlint ./(group).[foo].md

Error: Not found target files
```

_Originally posted by @azu in https://github.com/secretlint/secretlint/pull/1056#discussion_r2125985117_

**Comment by azu** [comment]
Currently, User can avoid this issue by pass escaped path to secretlint.

**Comment by tbashiyy** [comment]
@azu
Hello. I'm currently facing this issue.

> Currently, User can avoid this issue by pass escaped path to secretlint.

I use in sveltekit project.
sveltekit has file path like `src/routes/(app1)/users/[userId]`
(ref: https://svelte.jp/docs/kit/advanced-routing#Advanced-layouts-(group))

When using with `lint-staged`, lint-staged outputs file path without escaping. Then occurs this issue.

How do I handle this?🙇‍♂️

**Comment by tbashiyy** [comment]
Sorry. I could handle this by creating `lint-staged.config.js` to escape `(` `)` in my project.

https://github.com/lint-staged/lint-staged?tab=readme-ov-file#typescript

```javascript
/**
 * @type {import('lint-staged').Configuration}
 */
export default {
  '**/*': (files) => {
    const escapedFiles = files.map((file) =>
      file.replaceAll('(', String.raw`\(`).replaceAll(')', String.raw`\)`),
    );

    return `pnpm exec secretlint ${escapedFiles.join(' ')}`;
  },
};
``` 

I'm sorry for the trouble.🙇‍♂️

**Comment by azu** [comment]
As a fundamental problem, I think there is probably no way to make glob and escape compatible.
So I think we need a flag that allows secretlint to pass a path that doesn't interpret as a glob, like `--literal-path` or `--no-glob`.

**Comment by wbern** [comment]
**AI generated output reviewed by @wbern**

For anyone encountering "Not found target files" with lint-staged and files containing `()` or `[]` characters:

**Working solution:**
```js
"*": (filenames) =>
  filenames.length > 0
    ? `secretlint ${filenames
        .map((f) =>
          `"${f
            .replaceAll("(", String.raw`\(`)
            .replaceAll(")", String.raw`\)`)
            .replaceAll("[", String.raw`\[`)
            .replaceAll("]", String.raw`\]`)}"`,
        )
        .join(" ")}`
    : [],
```

This escapes glob-special characters before passing paths to secretlint, allowing it to treat them as literal filenames. Tested with Next.js route files like `app/(root)/[id]/page.tsx`.
